python/cudf/cudf/core/column/column.py

```diff
@@ -649,14 +652,28 @@ def append(self, other):
         return ColumnBase._concat([self, as_column(other)])
 
     def quantile(self, q, interpolation, exact):
-        if isinstance(q, Number):
+
+        is_number = isinstance(q, Number)
+
+        if is_number:
             quant = [float(q)]
         elif isinstance(q, list) or isinstance(q, np.ndarray):
             quant = q
         else:
             msg = "`q` must be either a single element, list or numpy array"
             raise TypeError(msg)
-        return libcudf.quantile.quantile(self, quant, interpolation, exact)
+
+        # get sorted indicies and exclude nulls
+        ordered_indices = cpp_sorted_order(self.as_frame(), [True], 1)
```

Comment:
Can we do just `self.sort_by_values()` here?

Reply:
`sort_by_values` create gathers a new series which we don't need. You mentioned `get_sorted_inds` elsewhere, and that looks friendlier, so I'll give that a shot.

Reply:
It's not obvious how to consolidate this, as it's used by both `DataFrame` and `Column`. Maybe just call it? :man_shrugging: